frontend/src/App.js

```diff
@@ -22,43 +65,136 @@ function App() {
       }
     };
 
-    // request local stream if requireLocalStream and localStream was not yet requested. 
     if (connection?.userType === "participant" && !localStream) {
       asyncStreamHelper();
     }
   }, [localStream, connection]);
 
   useEffect(() => {
-    const userType = "participant" // window.location.pathname === "/experimentRoom" ? "experimenter" : "participant";
-    const sessionId = "bbbef1d7d0";
-    const participantId = "aa798c85d5";
-    // Could be in useState initiator, but then the connection is executed twice / six times if in StrictMode. 
-    const newConnection = new Connection(userType, sessionId, participantId, true)
-    setConnection(newConnection)
+    const userType = "experimenter";
+    const newConnection = new Connection(userType, "", "", true);
+    setConnection(newConnection);
+
+    newConnection.start();
+    return () => {
+      newConnection.stop();
+    };
```

Comment:
`stop` will not be called if the browser is closed. To properly close the connection, use the `beforeunload` event on `window`. This could look something like:
```
useEffect(() => {
  // ...

  const closeConnection = () => {
    connection?.stop();
  }
  
  newConnection.start();
  window.addEventListener('beforeunload', closeConnection);
  return () => {
    window.removeEventListener('beforeunload', closeConnection);
	closeConnection();
  }
}, [connection]);
```